src/core/environment.js

```diff
@@ -1137,6 +1137,119 @@ function exitFullscreen() {
   }
 }
 
+
+/**
+ * Converts 3D world coordinates to 2D screen coordinates.
+ *
+ * This function takes a 3D vector and converts its coordinates
+ * from the world space to screen space. This is useful for placing
+ * 2D elements in a 3D scene or for determining the screen position
+ * of 3D objects.
+ *
+ * @method worldToScreen
+ * @param {p5.Vector} worldPosition The 3D coordinates in the world space.
+ * @return {p5.Vector} A vector containing the 2D screen coordinates.
+ * @example
+ * <div>
+ * <code>
+ * // Example 1: Convert 2D world coordinates of a rotating square to screen coordinates
+ * function setup() {
+ *   createCanvas(100, 100);
+ *
+ *   let vertices = [
+ *     createVector(-5, -5),
+ *     createVector(5, -5),
+ *     createVector(5, 5),
+ *     createVector(-5, 5)
+ *   ];
+ *
+ *   push(); // Start a new drawing state
+ *   translate(50, 50);
+ *   rotate(PI / 4);
+ *
+ *   // Convert each vertex to screen coordinates
+ *   let screenPos = vertices.map(v => worldToScreen(v));
+ *   pop(); // Restore original drawing state
+ *
+ *   background(200);
+ *
+ *   screenPos.forEach((pos, i) => {
+ *     text(`(${pos.x.toFixed(1)}, ${pos.y.toFixed(1)})`, pos.x, pos.y);
+ *   });
+ * }
+ * </code>
+ * </div>
+ * @example
+ * <div>
+ * <code>
+ * // Example 2: Convert 3D world coordinates of a rotating cube to 2D screen coordinates
+ * let vertices;
+ *
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   vertices = [
+ *     createVector(-25, -25, -25),
+ *     createVector(25, -25, -25),
+ *     createVector(25, 25, -25),
+ *     createVector(-25, 25, -25),
+ *     createVector(-25, -25, 25),
+ *     createVector(25, -25, 25),
+ *     createVector(25, 25, 25),
+ *     createVector(-25, 25, 25)
+ *   ];
+ * }
+ *
+ * function draw() {
+ *   background(200);
+ *
+ *   // Animate rotation
+ *   let rotationX = millis() / 1000;
+ *   let rotationY = millis() / 1200;
+ *
+ *   rotateX(rotationX);
+ *   rotateY(rotationY);
+ *
+ *   // Convert world coordinates to screen coordinates
+ *   let screenPos = vertices.map(v => worldToScreen(v));
+ *
+ *   // Display screen coordinates
+ *   screenPos.forEach((pos, i) => {
+ *     fill(255);
+ *     noStroke();
+ *     ellipse(pos.x, pos.y, 3, 3); // Draw points as small ellipses
+ *     fill(0);
+ *     text(`(${pos.x.toFixed(1)}, ${pos.y.toFixed(1)})`, pos.x, pos.y);
```

Comment:
Are the screen coordinates relative to the top left? if so we may need to subtract width/2 and height/2 to keep these centered when drawing them in WebGL mode